src/main/java/com/googlecode/lanterna/terminal/swing/AWTTerminalFontConfiguration.java

```diff
@@ -70,61 +76,54 @@
-                new Font("WenQuanYi Zen Hei", Font.PLAIN, getFontSize()),
-                new Font("AR PL UMing TW", Font.PLAIN, getFontSize()),
-                new Font("AR PL UMing HK", Font.PLAIN, getFontSize()),
-                new Font("AR PL UMing CN", Font.PLAIN, getFontSize())));
+                new Font("Ubuntu Mono", Font.PLAIN, fontSize),
+                new Font("FreeMono", Font.PLAIN, fontSize),
+                new Font("Liberation Mono", Font.PLAIN, fontSize),
+                new Font("VL Gothic Regular", Font.PLAIN, fontSize),
+                new Font("NanumGothic", Font.PLAIN, fontSize),
+                new Font("WenQuanYi Zen Hei Mono", Font.PLAIN, fontSize),
+                new Font("WenQuanYi Zen Hei", Font.PLAIN, fontSize),
+                new Font("AR PL UMing TW", Font.PLAIN, fontSize),
+                new Font("AR PL UMing HK", Font.PLAIN, fontSize),
+                new Font("AR PL UMing CN", Font.PLAIN, fontSize)));
     }
 
     private static List<Font> getDefaultFonts() {
+        int fontSize = getFontSize();
         return Collections.unmodifiableList(Collections.singletonList(
-                new Font("Monospaced", Font.PLAIN, getFontSize())));
+                new Font("Monospaced", Font.PLAIN, fontSize)));
     }
 
     // Here we check the screen resolution on the primary monitor and make a guess at if it's high-DPI or not
-    private static Integer CHOSEN_FONT_SIZE = null;
-    private synchronized static int getFontSize() {
-        if(CHOSEN_FONT_SIZE != null) {
-            return CHOSEN_FONT_SIZE;
-        }
-        // Source: http://stackoverflow.com/questions/3680221/how-can-i-get-the-monitor-size-in-java
-        GraphicsEnvironment ge      = GraphicsEnvironment.getLocalGraphicsEnvironment();
-        GraphicsDevice[]    gs      = ge.getScreenDevices();
-
-        // Assume the first GraphicsDevice is the primary screen (this isn't always correct but what to do?)
-        // Warning, there could be printers coming back here according to JavaDoc! Hopefully Java is reasonable and
-        // passes them in after the real monitor(s).
-        if (gs.length > 0) {
-            int primaryMonitorWidth = gs[0].getDisplayMode().getWidth();
-
-            // If the width is wider than Full HD (1080p, or 1920x1080), then assume it's high-DPI
-            if (primaryMonitorWidth > 2000) {
-                CHOSEN_FONT_SIZE = 28;
+    private static int getFontSize() {
+        if (Toolkit.getDefaultToolkit().getScreenResolution() >= 110) {
+            // Rely on DPI if it is a high value.
+            return Toolkit.getDefaultToolkit().getScreenResolution() / 7 + 1;
+        } else {
+            // Otherwise try to guess it from the monitor size:
+            // If the width is wider than Full HD (1080p, or 1920x1080), then assume it's high-DPI.
+            GraphicsEnvironment ge = GraphicsEnvironment.getLocalGraphicsEnvironment();
+            if (ge.getMaximumWindowBounds().getWidth() > 4000) {
```

Comment:
Since we're not using screen-resolutions but MaximumWindowBounds, I think 4000 is an unfortunate value.  Someone with a 4096 monitor then might see different font-sizes in lanterna apps based on how wide his sidebar is. E.g.: 4096 with a 96 pixel wide sidebar will make a smaller font than with an even-slightly-more-than-96 pixel wide sidebar.

If there isn't a convenient "the widest one of all the monitors" method, I'd propose 3840 as a somewhat "sweeter" value.

Reply:
Here's a chart of standard monitor sizes:  https://en.wikipedia.org/wiki/Display_resolution#/media/File:Vector_Video_Standards8.svg

Reply:
Yes, I completely forgot that somebody might have a sidebar. I have reduced it to >3840 and >1920 as suggested.

Reply:
I thought about it again and I think that 2k shouldn't be high-DPI yet. So a threshold of >4096 and >2048 should be better? That would give 512px for the sidebar for UWHD/WQHD/WXGA/QSXGA.